ebnf/main.go

```diff
@@ -0,0 +1,96 @@
+package main
+
+import (
+	"fmt"
+	"log"
+	"os"
+)
+
+type scanner struct {
+	input []byte
+	ch    byte
+	pos   int
+
+	insideBraces int
+}
+
+func (s *scanner) scan() {
+	s.skipUntilStart()
+	s.readByte() // consume %
+	s.readByte() // consume %
+
+	for {
+		if s.ch == 0 { // end
+			break
+		}
+
+		if s.ch == '{' {
+			s.insideBraces++
+			s.readByte()
+			continue
+		}
+
+		if s.ch == '}' {
+			s.insideBraces--
+			s.readByte()
+			continue
+		}
+
+		if s.insideBraces > 0 || s.ch == '%' || (s.ch == ';' && s.peekByte() != '\'') {
```

Comment:
If we are inside `Go` code we do nothing. Just consume the bytes. 

If we see `%` or `;` that is not next to single quote we do nothing as well. Just consume the bytes.